src/mvi/gpgmm.h

```diff
@@ -45,6 +45,9 @@ namespace gpgmm {
         MemoryAllocatorStats mStats = {};
     };
 
+    // MemoryAllocation represents a range of memory. A MemoryAllocation object will be held alive
```

Comment:
There is no guarantee MemoryAllocation will be held alive. So, "should be held alive" is more accurate.  Since a MemoryAllocation is only constructed to become a ResourceAllocation, WDYT about having ResourceAllocation get ref'd by the ResourceAllocator to enforce this?

Reply:
What exactly is the scenario where MemoryAllocation is not held alive?

Is the application not responsible for maintaining references to a ResourceAllocation? Wouldn't that be a memory leak if they didn't?

Reply:
When a ResourceAllocation needs to be constructed, a temporary MemoryAllocation is created then immediately destroyed without calling DeallocateMemory on it.

Once the ResourceAllocation is constructed from the temporary MemoryAllocation, then the allocation is "held alive" since it becomes a ref-counted type (where DeallocateMemory gets called automatically). The application developer cannot construct  or access a MemoryAllocation directly so if anything leaks there, it's the implementation's fault.